receiver/prometheusreceiver/internal/prom_to_otlp.go

```diff
@@ -50,12 +50,11 @@ func CreateNodeAndResourcePdata(job, instance, scheme string) *pdata.Resource {
 	attrs := resource.Attributes()
 	attrs.UpsertString(conventions.AttributeServiceName, job)
 	if isDiscernibleHost(host) {
-		attrs.UpsertString(conventions.AttributeHostName, host)
+		attrs.UpsertString(conventions.AttributeNetHostName, host)
```

Comment:
does this require a breaking change note in the changelog?

Reply:
Yes.  Right now, i've covered this with:

`prometheusreceiver` Produces different resource attributes when converting metrics to OTLP (#8266)

Do you have a different suggestion?

Reply:
I'd suggest an item in the breaking changes section calling out the rename from `host.name` to `net.host.name` to make it explicit for any users.

Reply:
updated to explicitly include each resource attribute changed